src/components/WalletIntegration/Web3ConnectManual.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react'
+import Web3Connect from 'web3connect'
+
+// Components
+import Button from '../Button'
+import DisplayWeb3 from '../DisplayWeb3'
+
+// API
+import getWeb3Api from '../../api/web3'
+
+import { SAFE_WALLETCONNECT_BRIDGE } from '../../globals'
+
+// Types
+import { Web3Api, provider as ProviderType } from '../../types'
+
+const web3Connect = new Web3Connect.Core({
+    providerOptions: {
+        walletconnect: {
+            bridge: SAFE_WALLETCONNECT_BRIDGE,
+        },
+    },
+})
+
+// subscribe to close
+web3Connect.on('close', () => {
+    console.log('Web3Connect Modal Closed') // modal has closed
+})
+
+const ManualConnector: React.FC = () => {
+    const [web3, setWeb3] = useState<Web3Api>()
+
+    useEffect(() => {
+        web3Connect.on('connect', async (provider: ProviderType) => {
+            try {
+                console.log(
+                    'Web3Connect.Core::Connected to',
+                    typeof provider === 'object' ? provider.constructor.name : provider,
+                )
+                const web3 = await getWeb3Api({ customProvider: provider }) // add provider to web3Api
+                setWeb3(web3)
+            } catch (error) {
+                console.error(error)
+            }
+        })
+    }, [])
+
+    return (
+        <>
+            <Button onClick={() => web3Connect.toggleModal()} secondary>
```

Comment:
Not sure 100% sure, but wouldn't be `onClick={web3Connect.toggleModal}` just enough?

Reply:
Agreed, `toggleModal` doesn't expect any arguments, so passing a `MouseEvent` is harmless